pom.xml

```diff
@@ -32,6 +32,7 @@
     <module>tensorflow-tools</module>
     <module>tensorflow-core</module>
     <module>tensorflow-training</module>
+    <module>tensorflow-frameworks</module>
```

Comment:
What do you think guys, should we move the dataset API to `tensorflow-training`? For me it make sense but I might not have all use cases in mind

Reply:
We could do that! I was thinking that it would be good to have the distinct frameworks exposed as separate artifacts, so that the artifact names mirror known pieces of the python framework... But that's not a strong preference, I could move it to `tensorflow-training`

Reply:
The users might not understand why they have to include so many different artifacts in their application, unless one is transitive from the other (e.g. `tensorflow-training` might depend on `tensorflow-data`). 

So I'm tempted to think to move everything in `tensorflow-training` right away, @Craigacp what do you think?

Reply:
@karllessard, I think that's reasonable. We can always split it out again if necessary.

Reply:
I think that makes sense, I'll move this `tensorflow-training`.

A question on the naming though—is it named `tensorflow-training` to make a hard distinction between training and inference? If so datasets (and other features we add in the future) will be used for both training and inference, so further down the road maybe it makes sense to consider the naming of this package...

Reply:
> is it named tensorflow-training to make a hard distinction between training and inference?

That was the original idea, yes. We want `tensorflow-core-api` to be as closest to the TF runtime as we can, and move higher-level APIs in other artifacts. 

I was foreseeing having Keras in its own artifact (`tensorflow-keras`) but having other API that fits more « in the middle » either in `tensorflow-training` or `tensorflow-inference`, so it is easy for a user to pick the right artifact based on the purpose of his application (having too many artifacts might confuse them more than helping them).

Now if you say that this Data API will be used as much for inference (outside Keras) than for training, the name might be wrong...  maybe then we should have all middle-level APIs in the same artifact (`tensorflow-framework`)? So packages would be `org.tensorflow.framework.training` and `org.tensorflow.framework.data`?

In the new proto packages though, there is already `org.tensorflow.proto.framework`, so it might be confusing if they do not relate on the same features

Reply:
This top is is still left opened, what do you think @Craigacp ?

Reply:
I think I'm ok with training moving into framework along with this data api. The proto package is a mess already so I'm happy documenting it as its own namespace and saying people shouldn't draw inferences from it's layout to the rest of the library.

Reply:
So @dhruvrajan , I think you are good to rename `tensorflow-training` by `tensorflow-framework` and change the package names so that they all begin with `org.tensorflow.framework` in that artifact.

Reply:
This is done!